### PR TITLE
Add labels to deployment, statefulset and service accounts

### PR DESCRIPTION
**Pull request by hamza3202**
This PR adds user defined labels to service accounts, deployment and stateful set. Existing `.Values.server.extraLabels` and `.Values.injector.extraLabels` are now added to more resources

### CODE REVIEW

**Comment by hamza3202** [comment]
@jasonodonnell Can you please review this PR?

**Review by jasonodonnell — COMMENTED** [review]
Hi @hamza3202, thanks for the contribution!

This looks great but there needs to be unit test coverage on all configurables. You can find the bats unit tests in `test/unit`. Let me know if you have any questions!

File: templates/server-statefulset.yaml [hunk 1]
```diff
@@ -11,6 +11,9 @@ metadata:
     app.kubernetes.io/name: {{ include "vault.name" . }}
     app.kubernetes.io/instance: {{ .Release.Name }}
     app.kubernetes.io/managed-by: {{ .Release.Service }}
+    {{- if  .Values.server.extraLabels -}}
```
Comment (jasonodonnell) [inline templates/server-statefulset.yaml:14]:
```suggestion
    {{- if .Values.server.extraLabels -}}
```

File: templates/server-serviceaccount.yaml [hunk 2]
```diff
@@ -11,6 +11,9 @@ metadata:
     app.kubernetes.io/name: {{ include "vault.name" . }}
     app.kubernetes.io/instance: {{ .Release.Name }}
     app.kubernetes.io/managed-by: {{ .Release.Service }}
+    {{- if  .Values.server.extraLabels -}}
```
Comment (jasonodonnell) [inline templates/server-serviceaccount.yaml:14]:
```suggestion
    {{- if .Values.server.extraLabels -}}
```

File: templates/injector-serviceaccount.yaml [hunk 3]
```diff
@@ -8,4 +8,7 @@ metadata:
     app.kubernetes.io/name: {{ include "vault.name" . }}-agent-injector
     app.kubernetes.io/instance: {{ .Release.Name }}
     app.kubernetes.io/managed-by: {{ .Release.Service }}
+    {{- if  .Values.injector.extraLabels -}}
```
Comment (jasonodonnell) [inline templates/injector-serviceaccount.yaml:11]:
```suggestion
    {{- if .Values.injector.extraLabels -}}
```

**Comment by Iristyle** [comment]
Just came here to file this same pull request!

Without the labels being applied to the `statefulset` and the `deployment` some of the utility is lost -- i.e. I can't use an app specific label across all resources for an application and key off that to purge resources, etc.

If these changes are ok and just need the tests, I can add those quick an open a new PR @jasonodonnell ?

**Comment by hamza3202** [comment]
@Iristyle If you could add the tests then that would be very good since I have been having a hard time trying to find time for this.

**Comment by Iristyle** [comment]
Can be closed in favor of #680... sorry it took so long to loop back, but we've got other areas of interest in this chart that we're looking to land for the next release so figured I'd fix this as well.

**Comment by hsimon-hashicorp** [comment]
Thanks for letting us know, @Iristyle! We'll look for you over on #680. :)

**Pull request closed without merging**